study1/week3/best-time-to-buy-and-sell-stock.java

```diff
@@ -0,0 +1,25 @@
+// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
+// Time Complexity: O(n), n: prices.length
+// Space Complexity: O(1), because 3 int-variables are declared -> O(1)+O(1)+O(1)
+class Solution {
+    public int maxProfit(int[] prices) {
+        int buyPrice = prices[0];
+        int sellPrice = 0;
+        int profit = sellPrice - buyPrice;  // store maxProfit
+
+        for (int i = 1; i < prices.length; ++i) {
```

Comment:
You can simplify this loop with `Math.min` and `Math.max` : 
```java
for (int i = 1; i < prices.length; ++i) {
    buyPrice = Math.min(buyPrice, prices[i]);
    profit = Math.max(profit, prices[i] - buyPrice);
}
```
it's computationally equivalent, but will be less code and easier to read.